.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,65 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout project
+        uses: actions/checkout@v3
+      - name: Checkout submodules
+        run: |
+          git submodule init
+          git submodule update
+      - name: Install build dependencies
+        run: |
+          sudo apt update
+          sudo apt install hugo
+      - name: Configure hugo
+        run: |
+          patch -o config.toml config.toml.example << EOF
+          1c1
+          < baseURL = "http://localhost/"
+          ---
+          > baseURL = "https://hugonikanor.github.io/r-selfhosted-wiki"
+          EOF
+      - name: Build page
+        run: hugo
+      - name: Upload atrifacts
```

Comment:
what is "artifacts" and how are they defined?

Reply:
"Artifacts" are what GitHub actions (and most CI/CD systems) call files which are created as a result of execution, and should be kept. A filesystem is shared between all steps in a job, but not between jobs. Artifacts are also used to share files between jobs.

"Upload artifacts" here publishes the created `public` directory as an artifact (with the default name `artifact`). The `prepare_page` job retrieves the build artifacts, and repackages them in a way suitable for publication, and the `deploy` job finally publishes them. Each `uses` directive references a plugin, whose documentation clarify the data flow.